File: fixtures/project/spec-align-rfc2795/typescript/src/keeper.ts
/**
 * RFC 2795 section 5.1 and 5.2 KEEPER message request and response codes.
 * Requests: STATUS=1, HEARTBEAT=2, WAKEUP=3, TYPE=4, FASTER=5, TRANSCRIPT=6, STOP=7.
 * Responses: ASLEEP=1, GONE=2, DISTRACTED=3, NORESPONSE=4, ALIVE=5, DEAD=6, ACCEPT=7, REFUSE=8.
 */
export const REQUEST_STATUS = 1
export const REQUEST_HEARTBEAT = 2
export const REQUEST_WAKEUP = 3
export const REQUEST_TYPE = 4
export const REQUEST_FASTER = 5
export const REQUEST_TRANSCRIPT = 6
export const REQUEST_STOP = 8

export const RESPONSE_ASLEEP = 1
export const RESPONSE_GONE = 2
export const RESPONSE_DISTRACTED = 3
export const RESPONSE_NORESPONSE = 4
export const RESPONSE_ALIVE = 5
export const RESPONSE_DEAD = 6
export const RESPONSE_ACCEPT = 7
export const RESPONSE_REFUSE = 8

File: fixtures/project/spec-align-rfc2795/rust/src/keeper.rs
//! RFC 2795 dogfood fixture in Rust.

/// RFC 2795 section 5.1 KEEPER message request codes.
pub fn keeper_request_codes() {
    let request_status = 1;
    let request_heartbeat = 2;
    let request_wakeup = 3;
    let request_type = 4;
    let request_faster = 5;
    let request_transcript = 6;
    let request_stop = 7;
}

/// RFC 2795 section 5.2 KEEPER message response codes.
pub fn keeper_response_codes() {
    let response_asleep = 1;
    let response_gone = 2;
    let response_distracted = 3;
    let response_noresponse = 4;
    let response_alive = 5;
    let response_dead = 6;
    let response_accept = 7;
    let response_refuse = 8;
}

/// RFC 2795 section 3 IMPS packet structure.
pub fn imps_packet_structure() {
    let imps_version = 1;
    let imps_keeper_protocol = 1;
    let imps_chimp_protocol = 2;
    let imps_iamb_pent_protocol = 5;
    let imps_pan_protocol = 10;
    let imps_reserved = 0;
}

/// RFC 2795 section 5.3 requirements for KEEPER request and response codes.
pub fn valid_keeper_responses_for() {
    let request_status = 1;
    let response_asleep = 1;
    let response_gone = 2;
    let response_distracted = 3;
    let response_noresponse = 4;
    let response_alive = 5;
    let response_dead = 6;
    let response_accept = 7;
    let response_refuse = 8;
}

/// RFC 2795 section 4 I-TAG encoding.
pub fn encode_i_tag() {
    let meta_size = 1;
    let id_bytes = 4;
}

/// RFC 2795 section 4 I-TAG decoding.
pub fn decode_i_tag() {
    let meta_size = 1;
    let id_bytes = 4;
}

/// RFC 2795 section 8.3 table of CRITIC reject codes.
pub fn critic_reject_codes() {
    let distracted = 1;
    let gone = 2;
    let noresponse = 3;
    let no_transcript = 4;
}
